frontend/src/utils/format.ts

```diff
@@ -104,6 +113,37 @@ export const convertTimestampToRelative = (
 		addSuffix,
 	})
 
+/**
+ * Calculates effective time of the event.
+ * When event effectively happens on the next payday after the event
+ * @param {string} timestamp Time of the event
+ * @param {string} nextPaydayTime Time of the next payday. This should be take from the most recent block ideally.
+ * @returns {string} Effective time of the event. Time when event will actually take place.
+ */
+export const tillNextPayday = (timestamp: string, nextPaydayTime: string) => {
+	const time = parseISO(timestamp)
+	const paydayTime = parseISO(nextPaydayTime)
+	const nextDay = addDays(time, 1)
```

Comment:
Ok, so this is going to work at the moment, but it is not entirely correct.

In order for it to be correct you'd have to get the paydayDuration from the chain. paydayDuration is not fixed, and is a chain parameter. At the moment it is 1 day, and will likely be 1 day for the foreseeable future.

Is that very hard to do?

Reply:
Should not be @abizjak 
Can you help us with a property thats returned by json having the information for payday duration and not next payday time? is `rewardPeriodLength` is good example?

Reply:
Yes, it is rewardPeriodLength. I think that is in "epochs" so you need to convert it to seconds.

Reply:
fixed